.github/workflows/push-mainline-to-github-pages.yml

```diff
@@ -1,26 +1,34 @@
-# This workflow is necessary to support publishing to GitHub pages after renaming master to mainline since GitHub pages currently only support setting source from master branch
+# This workflow is necessary to support publishing to GitHub pages as it will push the docs folder to gh-pages.
 
-name: Push mainline changes into master branch
+name: Push mainline changes into gh-pages branch
 
 # Controls when the action will run. Triggers the workflow on push or pull request
-# events but only for the master branch
+# events but only for the gh-pages branch
 on:
   push:
     branches: [mainline]
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  push_to_master_branch:
+  push_to_gh-pages_branch:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
```

Comment:
nit: From this page: it looks like the `action/setup-node` has already updated to v3.

Reply:
Updated to v3 for checkout and setup-node for all of our GH Actions in latest commit.